src/api/WebLink.js

```diff
@@ -28,6 +28,44 @@ class WebLink extends Item {
         const suffix: string = id ? `/${id}` : '';
         return `${this.getBaseApiUrl()}/web_links${suffix}`;
     }
+
+    /**
+     * Gets a box weblink.
+     *
+     * @param {string} id - Weblink id
+     * @param {Function} successCallback - Function to call with results
+     * @param {Function} errorCallback - Function to call with errors
+     * @param {Array<String>} fields - Array of field strings
+     * @returns {Promise}
+     */
+    async getWeblink(
+        id: string,
+        successCallback: Function,
+        errorCallback: Function,
+        { fields }: FetchOptions = {},
+    ): Promise<void> {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const cacheKey = this.getCacheKey(id);
+        const params = { fields: fields.toString() };
+
+        try {
+            const { data } = await this.xhr.get({
+                url: this.getUrl(id),
+                params,
+                headers: { 'X-Rep-Hints': X_REP_HINTS },
+            });
+
+            const cachedEntry = this.getCache().get(cacheKey);
+            const updatedCacheEntry = { ...cachedEntry, ...data };
```

Comment:
Should this do a deep merge rather than a shallow assignment?

Reply:
right